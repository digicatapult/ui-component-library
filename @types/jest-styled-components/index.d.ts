declare module 'jest-styled-components/serializer'
