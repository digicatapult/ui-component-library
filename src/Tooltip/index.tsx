import React, { PropsWithChildren } from 'react'
import styled from 'styled-components'

export interface TooltipProps {
  background?: string
  padding?: string
  margin?: string
  width?: string
  height?: string
  borderRadius?: string
  leftOrRight?: string
}

const Tooltip: React.FC<PropsWithChildren<TooltipProps>> = ({
  children,
  background = '#f0f0f0',
  padding = '1em',
  margin = '0',
  width = 'auto',
  height = 'auto',
  borderRadius = '1em',
  leftOrRight = 'right',
}) => {
  return (
    <Wrapper
      background={background}
      padding={padding}
      margin={margin}
      width={width}
      height={height}
      borderRadius={borderRadius}
      leftOrRight={leftOrRight}
    >
      {children}
    </Wrapper>
  )
}

const Wrapper = styled.div<TooltipProps>`
  background: ${({ background }) => background};
  padding: ${({ padding }) => padding};
  width: ${({ width }) => width};
  height: ${({ height }) => height};
  margin: ${({ margin }) => margin};
  border-radius: ${({ borderRadius }) => borderRadius};
  box-shadow: 0px 4px 5px rgba(0, 0, 0, 0.1);
  position: relative;
  box-sizing: border-box;

  ::before {
    content: '';
    position: absolute;
    width: 0;
    height: 0;
    bottom: -0.7em;
    border: 0.8em solid ${({ background }) => background};
    rotate: 45deg;
    border-bottom-right-radius: 0.8em;
    box-shadow: 4px 4px 5px rgba(0, 0, 0, 0.1);
    ${({ leftOrRight }) => leftOrRight}: 2em;
  }
`

export default Tooltip
