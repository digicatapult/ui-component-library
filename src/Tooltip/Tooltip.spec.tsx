import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Tooltip from './index.js'

describe('Tooltip', () => {
  test('default', () => {
    const tree = renderer.create(<Tooltip>Some content</Tooltip>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('overrides', () => {
    const tree = renderer
      .create(
        <Tooltip width="28ch" background="#FFF" leftOrRight="left">
          Some content
        </Tooltip>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
