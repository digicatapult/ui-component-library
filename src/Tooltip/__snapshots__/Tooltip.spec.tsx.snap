// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip default 1`] = `
.c0 {
  background: #f0f0f0;
  padding: 1em;
  width: auto;
  height: auto;
  margin: 0;
  border-radius: 1em;
  box-shadow: 0px 4px 5px rgba(0,0,0,0.1);
  position: relative;
  box-sizing: border-box;
}

.c0::before {
  content: '';
  position: absolute;
  width: 0;
  height: 0;
  bottom: -0.7em;
  border: 0.8em solid #f0f0f0;
  rotate: 45deg;
  border-bottom-right-radius: 0.8em;
  box-shadow: 4px 4px 5px rgba(0,0,0,0.1);
  right: 2em;
}

<div
  className="c0"
  height="auto"
  width="auto"
>
  Some content
</div>
`;

exports[`Tooltip overrides 1`] = `
.c0 {
  background: #FFF;
  padding: 1em;
  width: 28ch;
  height: auto;
  margin: 0;
  border-radius: 1em;
  box-shadow: 0px 4px 5px rgba(0,0,0,0.1);
  position: relative;
  box-sizing: border-box;
}

.c0::before {
  content: '';
  position: absolute;
  width: 0;
  height: 0;
  bottom: -0.7em;
  border: 0.8em solid #FFF;
  rotate: 45deg;
  border-bottom-right-radius: 0.8em;
  box-shadow: 4px 4px 5px rgba(0,0,0,0.1);
  left: 2em;
}

<div
  className="c0"
  height="auto"
  width="28ch"
>
  Some content
</div>
`;
